nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2DataAccess.java

```diff
@@ -13,124 +13,62 @@
-import static org.sonatype.nexus.repository.storage.MetadataNodeEntityAdapter.P_NAME;
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Shared code between P2 facets.
  */
 @Named
 public class P2DataAccess
 {
-  public static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1);
+  final Logger log = Loggers.getLogger(this);
 
-  /**
-   * Find a component by its name and tag (version)
-   *
-   * @return found component of null if not found
-   */
-  @Nullable
-  public Component findComponent(final StorageTx tx,
-                                 final Repository repository,
-                                 final String name,
-                                 final String version)
-  {
-    Iterable<Component> components = tx.findComponents(
-        Query.builder()
-            .where(P_NAME).eq(name)
-            .and(P_VERSION).eq(version)
-            .build(),
-        singletonList(repository)
-    );
-    if (components.iterator().hasNext()) {
-      return components.iterator().next();
-    }
-    return null;
-  }
+  private final AttributesParserFeatureXml featureXmlParser;
 
-  /**
-   * Find an asset by its name.
-   *
-   * @return found asset or null if not found
-   */
-  @Nullable
-  public Asset findAsset(final StorageTx tx, final Bucket bucket, final String assetName) {
-    return tx.findAssetWithProperty(MetadataNodeEntityAdapter.P_NAME, assetName, bucket);
-  }
+  private final AttributesParserManifest manifestParser;
 
-  /**
-   * Save an asset && create blob.
-   *
-   * @return blob content
-   */
-  public Content saveAsset(final StorageTx tx,
-                           final Asset asset,
-                           final Supplier<InputStream> contentSupplier,
-                           final Payload payload) throws IOException
+  @Inject
+  public P2DataAccess(final AttributesParserFeatureXml featureXmlParser,
+                      final AttributesParserManifest manifestParser)
   {
-    AttributesMap contentAttributes = null;
-    String contentType = null;
-    if (payload instanceof Content) {
-      contentAttributes = ((Content) payload).getAttributes();
-      contentType = payload.getContentType();
-    }
-    return saveAsset(tx, asset, contentSupplier, contentType, contentAttributes);
+    this.featureXmlParser = checkNotNull(featureXmlParser);
+    this.manifestParser = checkNotNull(manifestParser);
   }
 
-  /**
-   * Save an asset && create blob.
-   *
-   * @return blob content
-   */
-  public Content saveAsset(final StorageTx tx,
-                           final Asset asset,
-                           final Supplier<InputStream> contentSupplier,
-                           final String contentType,
-                           @Nullable final AttributesMap contentAttributes) throws IOException
+  @VisibleForTesting
+  public P2Attributes mergeAttributesFromTempBlob(final TempBlob tempBlob, final P2Attributes sourceP2Attributes)
```

Comment:
why P2DataAccess, (should be something related to DB from the name) has only one function mergeAttributesFromTempBlob